nifi-nar-bundles/nifi-azure-bundle/nifi-azure-processors/src/main/java/org/apache/nifi/processors/azure/storage/utils/AzureBlobClientSideEncryptionUtils_v12.java

```diff
@@ -0,0 +1,125 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.nifi.processors.azure.storage.utils;
+
+import org.apache.commons.codec.DecoderException;
+import org.apache.commons.codec.binary.Hex;
+import org.apache.nifi.components.AllowableValue;
+import org.apache.nifi.components.PropertyDescriptor;
+import org.apache.nifi.components.ValidationContext;
+import org.apache.nifi.components.ValidationResult;
+import org.apache.nifi.expression.ExpressionLanguageScope;
+import org.apache.nifi.processor.util.StandardValidators;
+import org.apache.nifi.util.StringUtils;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.List;
+
+public class AzureBlobClientSideEncryptionUtils_v12 {
+
+    private static final String DEFAULT_KEY_ID = "nifi";
+    public static final int KeySize128 = 128 >> 3;
+    public static final int KeySize192 = 192 >> 3;
+    public static final int KeySize256 = 256 >> 3;
+    public static final int KeySize384 = 384 >> 3;
+    public static final int KeySize512 = 512 >> 3;
+
+    public static final PropertyDescriptor CSE_KEY_TYPE = new PropertyDescriptor.Builder()
+            .name("cse-key-type")
+            .displayName("Client-Side Encryption Key Type")
+            .required(true)
+            .allowableValues(buildCseEncryptionMethodAllowableValues())
+            .defaultValue(AzureBlobClientSideEncryptionMethod_v12.NONE.name())
+            .description("Specifies the key type to use for client-side encryption.")
+            .build();
+
+    public static final PropertyDescriptor CSE_KEY_ID = new PropertyDescriptor.Builder()
+            .name("cse-key-id")
+            .displayName("Client-Side Encryption Key ID")
+            .description("Specifies the ID of the key to use for client-side encryption.")
+            .expressionLanguageSupported(ExpressionLanguageScope.FLOWFILE_ATTRIBUTES)
+            .required(false)
+            .addValidator(StandardValidators.NON_BLANK_VALIDATOR)
+            .dependsOn(CSE_KEY_TYPE, AzureBlobClientSideEncryptionMethod_v12.LOCAL.name())
+            .build();
+
+    public static final PropertyDescriptor CSE_LOCAL_KEY_HEX = new PropertyDescriptor.Builder()
+            .name("cse-local-key-hex")
+            .displayName("Local Key")
+            .description("When using local client-side encryption, this is the raw key, encoded in hexadecimal")
+            .required(false)
+            .addValidator(StandardValidators.NON_BLANK_VALIDATOR)
+            .dependsOn(CSE_KEY_TYPE, AzureBlobClientSideEncryptionMethod_v12.LOCAL.name())
+            .sensitive(true)
+            .build();
```

Comment:
Instead of defining these property descriptors on the `Utils` class, recommend creating a new interface named `ClientSideEncryptionSupport` and adding the Property Descriptors on that interface. Then the implementing Processors and implement the `ClientSideEncryptionSupport` interface to access the properties.